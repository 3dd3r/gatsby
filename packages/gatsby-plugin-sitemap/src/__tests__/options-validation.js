import { pluginOptionsSchema } from "../options-validation"
import Joi from "joi"

const schema = pluginOptionsSchema({ Joi })

describe(`gatsby-plugin-sitemap: options-validation tests`, () => {
  describe(`validateOptions`, () => {
    it(`creates correct defaults`, async () => {
      const pluginOptions = await schema.validateAsync({})

      expect(pluginOptions).toMatchSnapshot()
    })

    it(`errors on invalid options`, async () => {
      try {
        await schema.validateAsync({ wrong: `test` })
      } catch (error) {
        expect(error).toMatchSnapshot()
      }
    })
  })
})
