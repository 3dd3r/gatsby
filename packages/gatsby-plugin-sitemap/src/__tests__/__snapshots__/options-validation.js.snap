// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-sitemap: options-validation tests validateOptions creates correct defaults 1`] = `
Object {
  "createLinkInHead": true,
  "entryLimit": 45000,
  "excludes": Array [],
  "filterPages": [Function],
  "output": "/sitemap",
  "query": "{ site { siteMetadata { siteUrl } } allSitePage { nodes { path } } }",
  "resolvePagePath": [Function],
  "resolvePages": [Function],
  "resolveSiteUrl": [Function],
  "serialize": [Function],
}
`;

exports[`gatsby-plugin-sitemap: options-validation tests validateOptions errors on invalid options 1`] = `[ValidationError: "wrong" is not allowed]`;
